README.md

```diff
@@ -203,7 +203,35 @@ COPY bin/my-wag-service /usr/bin/my-wag-service
 
   * If you don't have a context to pass to a Wag function you have two options
     * context.Background() - use this when this is the creator of the request chain, like a test or a top-level service.
-    * context.TODO() - use this when you haven't been passed a context from a caller yet, but you expect the caller to send you one at some point.  
+    * context.TODO() - use this when you haven't been passed a context from a caller yet, but you expect the caller to send you one at some point.
+
+
+### DynamoDB Codegen
+
+  * Wag can auto-generate server code to save models to DynamoDB if you specify the `x-db` extension on a schema:
+  ```yaml
+definitions:
+  Thing:
+    x-db:
+      AllowOverwrites: false
+      DynamoDB:
+        KeySchema:
+          - AttributeName: name
+            KeyType: HASH
+          - AttributeName: version
+            KeyType: RANGE
+    type: object
+    properties:
+      name:
+        type: string
+      version:
+        type: integer
+  ```
+  The above will generate a `db` package with code to load/persist `Thing` objects from/to DynamoDB.
+  * `AllowOverwrites` specifies whether the auto-generated `Save` method should succeed if the object already exists.
```

Comment:
Maybe we could eliminate the need for this config by having more methods on the DB interface that have specific behaviors.

- Create (fails if already exists)
- Update (fails if DNE yet)
- Upsert / CreateOrUpdate / etc (creates or updates as needed)

Reply:
After reading further, I also wonder whether having `AllowOverwrites` as a bool is flexible enough. 

I could imagine at the application level there might be some times when overwriting is OK and other times where it's not.

Reply:
For now I'm designing this mostly to target the `workflow-manager` use case. I worry about over-designing it with future / unknown use cases in mind, so I'd rather evolve the design in conjunction with real use cases. Right now the config is tailored for `WorkflowDefinition` objects, which cannot be overwritten, and I expect it to evolve as it expands to support other objects